Core.Collectors/Web/HttpClientWrapper.cs

```diff
@@ -28,27 +29,37 @@ public HttpClientWrapper()
             this.httpClient.DefaultRequestHeaders.ConnectionClose = false;
         }
 
-        public Task<HttpResponseMessage> GetAsync(string requestUrl, IAuthentication authentication)
-        {
-            return this.GetAsync(requestUrl, authentication, productInfoHeaderValue: null);
-        }
-
         public Task<HttpResponseMessage> GetAsync(string requestUrl, IAuthentication authentication, ProductInfoHeaderValue productInfoHeaderValue)
         {
             return this.GetAsync(requestUrl, authentication, productInfoHeaderValue, eTag: string.Empty);
         }
 
         public Task<HttpResponseMessage> PostAsync(string requestUrl, string requestBody, IAuthentication authentication, ProductInfoHeaderValue productInfoHeaderValue)
         {
-            return this.MakeRequestAsync(requestUrl, HttpMethod.Post, requestBody, authentication, productInfoHeaderValue, eTag: string.Empty);
+            return this.PostAsync(requestUrl, requestBody, authentication, productInfoHeaderValue, additionalHeaders: new Dictionary<string, string>());
```

Comment:
Super minor - the allocation of a new dictionary (despite a very small memory foot print) can be saved by using `null` and handling that in send logic but since you are using Azure Functions I guess this does not really matter (comparing to on a long-running server for GC). Feel free ignore

Reply:
Thanks, agreed that this should not be that much of an issue in an Azure Function. Whenever applicable, I prefer making things not possible to be null (to prevent NRE), rather than handle null, down the path. I guess you can say, here we are willing to pay performance penalty (however small it is) for mental simplification.